ramp-engine/rampbkd/dispatcher.py

```diff
@@ -141,20 +141,19 @@ def collect_result(self):
                 logger.info('Collecting results from worker {}'.format(worker))
                 returncode = worker.collect_results()
                 set_submission_state(
-                    self._database_config, submission_id,
+                    session, submission_id,
                     'tested' if not returncode else 'training_error'
                 )
                 self._processed_submission_queue.put_nowait(
                     (submission_id, submission_name))
                 worker.teardown()
 
-    def update_database_results(self):
+    def update_database_results(self, session):
         """Update the database with the results of ramp_test_submission."""
         while not self._processed_submission_queue.empty():
             submission_id, submission_name = \
                 self._processed_submission_queue.get_nowait()
-            if 'error' in get_submission_state(self._database_config,
-                                               submission_id):
+            if 'error' in get_submission_state(session, submission_id):
```

Comment:
We need to update all the leaderboard even if it is failing.

```suggestion
            if 'error' in get_submission_state(session, submission_id):
                update_leaderboards(self._ramp_config['event_name'])
                update_all_user_leaderboards(self._ramp_config['event_name'])
                set_submission_state(session, submission_id, 'training_error')
```